agent/agent_test.go

```diff
@@ -577,18 +593,34 @@ func TestAgent_ServiceWithTagChurn(t *testing.T) {
 		t.Fatalf("err: %v", err)
 	}
 
+	chk = &structs.HealthCheck{
+		CheckID: "node-check",
+		Name:    "Node-level check",
+		Status:  api.HealthCritical,
+	}
+	chkt = &structs.CheckType{
+		TTL: time.Hour,
+	}
+	if err := a.AddCheck(chk, chkt, true, ""); err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
 	if err := a.sync.State.SyncFull(); err != nil {
 		t.Fatalf("err: %v", err)
 	}
+
 	args := &structs.ServiceSpecificRequest{
 		Datacenter:  "dc1",
 		ServiceName: "redis",
 	}
-	var before structs.IndexedHealthChecks
-	if err := a.RPC("Health.ServiceChecks", args, &before); err != nil {
+	var before structs.IndexedCheckServiceNodes
```

Comment:
Why did the test have to change from calling Health.ServiceChecks to Health.ServiceNodes to illustrate this problem?

Reply:
This problem wasn't uncovered by this test since you actually have to hit specific HTTP API endpoints to corrupt the data, but I did add a node-level check as part of trying to chase this down, so I changed this RPC endpoint to one that returns the service's check + the node-level checks to make this test more robust. It was when I could see this test passing but run this in an integrated fashion and watching /v1/health/service/<foo> and see churn that I knew something super weird was happening!